src/goto-instrument/synthesizer/enumerative_loop_invariant_synthesizer.h

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************\
+
+Module: Enumerative Loop Invariant Synthesizer
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+/// \file
+/// Enumerative Loop Invariant Synthesizer
+
+#ifndef CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_ENUMERATIVE_LOOP_INVARIANT_SYNTHESIZER_H
+#define CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_ENUMERATIVE_LOOP_INVARIANT_SYNTHESIZER_H
+
+#include <goto-programs/goto_model.h>
+
+#include "loop_invariant_synthesizer_base.h"
+
+class messaget;
+
+/// Enumerative loop invariant synthesizers.
+/// It handles `goto_model` containing only checks instrumented by
```

Comment:
❓ I guess this comment is not yet relevant as it doesn't have this restriction yet, but a different one (only producing TRUE)?

Reply:
No, it doesn't have this restriction yet. This comment restrict the full implementation of `enumerative_loop_invariant_synthesizer` which contain in later PRs.